CI: Use Node.js version 20

jmillan opened the pull request.

jmillan committed:
CI: Use Node.js version 20

jmillan committed:
Update CHANGELOG.md

ibc reviewed (approved):
What about the Node version in the Docker file?

jmillan committed:
Update Dockerfile

Previous way to install Node is deprecated. Update to install Node
according to the docs: https://github.com/nodesource/distributions

jmillan commented:
For some reason CodeQL is complaining about not finding the C++ code..

As per the logs of the action it builds mediasoup correctly.

ibc commented:
> For some reason CodeQL is complaining about not finding the C++ code..
> 
> As per the logs of the action it builds mediasoup correctly.

I'd just ignore it. It cannot be related to this PR. We can investigate later and even remove it if whatever it complains about is not easy to fix.

jmillan committed:
CodeQL: skip worker prebuilt download

jmillan commented:
> I'd just ignore it. It cannot be related to this PR. We can investigate later and even remove it if whatever it complains about is not easy to fix.

As per the docs, it seems that the code needs to be compiled by the action (it seems weird to me, but..). I've just disabled the prebuilt download and giving it a try.

EDIT: [Docs]( https://docs.github.com/en/code-security/code-scanning/troubleshooting-code-scanning/no-source-code-seen-during-build)

jmillan committed:
Windows: try to fix the deletion of msys

jmillan commented:
> As per the docs, it seems that the code needs to be compiled by the action (it seems weird to me, but..). I've just disabled the prebuilt download and giving it a try.

That fixed it.

Now trying to fix a issue (present in V3 already) about not being able to remove `msys` on Windows.

jmillan committed:
Revert "Windows: try to fix the deletion of msys"

This reverts commit 8065234f42f10a47292a837fe21d2b1699bcbd98.

ibc commented:
> As per the docs, it seems that the code needs to be compiled by the action (it seems weird to me, but..). I've just disabled the prebuilt download and giving it a try.
> 
> EDIT: Docs

I've read those docs and I don't get it. What does it mean that you have disabled the prebuilt download and give it a try?

jmillan commented:
> I've read those docs and I don't get it. What does it mean that you have disabled the prebuilt download and give it a try?

The docs say the code needs to be compiled. I'm not trying to answer the question 'how was it working before'.

<img width="773" alt="Screenshot 2023-10-17 at 10 06 43" src="https://github.com/versatica/mediasoup/assets/732340/3b4ec291-73da-4ba3-a0bb-e1efc817974a">

ibc commented:
> The docs say the code needs to be compiled. 

Which code? Is this about prebuilt worker binaries or what? Also, we do obviously compile those...

jmillan commented:
> Which code? Is this about prebuilt worker binaries or what? Also, we do obviously compile those...

The C++ code.... We were not compiling it within the CodeQL workflow because we were downloading the prebuilt binaries instead. Try removing the change I added in the workflow file to see the logs.

ibc commented on .github/workflows/codeql.yml:
```diff
@@ -36,6 +36,9 @@ jobs:
         # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
 
+    env:
+      MEDIASOUP_SKIP_WORKER_PREBUILT_DOWNLOAD: "true"
```
Wow I completely missed this!
